hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/util/StreamerUtil.java

```diff
@@ -183,6 +183,7 @@ public static TypedProperties flinkConf2TypedProperties(Configuration conf) {
         properties.put(option.key(), option.defaultValue());
       }
     }
+    properties.put(HoodieTableConfig.TYPE.key(), conf.getString(FlinkOptions.TABLE_TYPE));
```

Comment:
Its foreign here, to fox broken testWriteMultiWriterPartialOverlapping you just need to remove HoodieTableConfig.TYPE property in test